app/services/ccms_user/user_details/silas.rb

```diff
@@ -0,0 +1,45 @@
+module CCMSUser
+  module UserDetails
+    class Silas
+      ApiError = Class.new(StandardError)
+      UserNotFound = Class.new(StandardError)
+
+      extend Forwardable
+
+      def self.call(silas_id)
+        new(silas_id).call
+      end
+
+      def initialize(silas_id)
+        @silas_id = silas_id
+      end
+
+      def call
+        response = request
+
+        if response.success?
+          JSON.parse(response.body)
+        elsif response.status == 404
+          provider = Provider.find_by(silas_id:)
+          Rails.logger.info("#{self.class} - No provider details found for #{provider.email}")
+          raise UserNotFound, "No CCMS username found for #{provider.email}"
+        else
+          raise ApiError, "API Call Failed: status #{response.status}, body #{response.body.presence || 'nil'}"
+        end
+      end
+
+    private
+
+      def_delegators :connection, :get
+      attr_reader :silas_id
+
+      def request
+        get("user-details/silas/#{silas_id}")
+      end
+
+      def connection
+        @connection ||= Connection.new
+      end
+    end
+  end
+end
```

Comment:
I assume you've done this namespace structure base on the endpoint structure. 

Is this just in case they expose future endpoints we might need / want to call? 

Just finding the 'Silas' part a bit confusing, rather than the class just being 'UserDetails'

Reply:
Yes, my thinking was so we _could_ extend to say `CCMSUser::UserDetails::CCMS` or `CCMSUser::UserDetails::User`.  but i did have it as just `CCMSUser::UserDetails` at first 
and changed because i was thinking possibly too much about extensibility.